### PR TITLE
Feature/persist s3 path

### PR DESCRIPTION
**Pull request by daiLlew**
### What

Changes to include the S3 URL for the uploaded file in the Kafka message.
Removed redundant http handlers.

### How to review

s3url should be included in the outbound kafka message.

### Who can review

@CarlHembrough @ian-kent @NeilMadden @cmcgarel @fenallen

### CODE REVIEW

File: config/uploader.go [hunk 1]
```diff
@@ -3,29 +3,88 @@ package config
...
 
 // TopicName the name of the Kafka topic to send messages to.
 var TopicName = "file-uploaded"
 
+var AWScfg *AWSConfig
+
+// Default S3 URL value.
+var s3URL, _ = url.Parse("s3://dp-csv-splitter")
+
+type AWSConfig struct {
```
Comment (ian-kent) [inline config/uploader.go]: package feels like its doing a lot more than just 'config' - the AWSConfig struct seems right here, but should all the other stuff (`extractURLPath`, `GetFilePath` etc) be somewhere else? I'm guessing they're called often as part of normal running of the app

File: config/uploader.go [hunk 2]
```diff
@@ -3,29 +3,88 @@ package config
...
+	bucketName *string
+	path       string
+	url        *url.URL
+	awsRegion  *string
```
Comment (ian-kent) [inline config/uploader.go]: don't think we need to handle `region` - afaik, as long as `AWS_REGION` is set, the aws library will read it in from there automatically (like it does with access key ID etc)

File: file/s3/store_test.go [hunk 3]
```diff
@@ -44,3 +49,62 @@ func TestUpload(t *testing.T) {
 		})
 	})
 }
+
+/*func TestFileStore_GetUploadInputKey(t *testing.T) {
```
Comment (ian-kent) [inline file/s3/store_test.go]: test should be removed if its not required

File: handler/upload.go [hunk 4]
```diff
@@ -64,10 +66,14 @@ func Upload(w http.ResponseWriter, req *http.Request) {
...
 
-	err = EventProducer.FileUploaded(event.FileUploaded{
-		Filename: header.Filename,
+	event := event.FileUploaded{
 		Time:     time.Now().UTC().Unix(),
-	})
+		S3URL: config.AWScfg.GetS3FileURL(header.Filename),
+	}
+
+	log.Debug("I AM SENDING " + fmt.Sprintf("%+v", event), nil)
```
Comment (ian-kent) [inline handler/upload.go]: :grinning:

Comment (ian-kent) [inline handler/upload.go]: edit: should probably be `log.Debug("sending file to S3", log.Data{"url": s3URL})`

no need to include time, that'll already be included by the logger

File: config/uploader.go (hunk 1 above)
Comment (daiLlew) [inline config/uploader.go]: My thought process was that those 'actions' belong to the model which is why I added them here.

File: file/s3/store_test.go (hunk 3 above)
Comment (daiLlew) [inline file/s3/store_test.go]: removed.

File: handler/upload.go (hunk 4 above)
Comment (daiLlew) [inline handler/upload.go]: Done.

File: config/uploader.go (hunk 2 above)
Comment (daiLlew) [inline config/uploader.go]: I thought it felt like it belongs with the other AWS config.

File: config/uploader.go (hunk 1 above)
Comment (ian-kent) [inline config/uploader.go]: I think we need a clearer line between 'config' and 'does stuff'

config should encapsulate the data retrieved from the config env vars, anything else feels like more than config and should be somewhere else - e.g., importing `"github.com/aws/aws-sdk-go/aws"` into a `config` package suggests the boundary is wrong

File: config/uploader.go (hunk 2 above)
Comment (ian-kent) [inline config/uploader.go]: i'd agree if it was needed, but I don't think we need to read the value in at all, or pass it around, or provide it to the aws sdk - imo we should rely on it reading the value from `AWS_REGION` like it does for `AWS_SECRET_ACCESS_KEY` and `AWS_ACCESS_KEY_ID`

**Pull request merged**